my_test_app.py

```diff
@@ -3,12 +3,51 @@
 from flask import Flask
 app = Flask(__name__)
 
-@app.route('/weather/odessa/')
-def hello_world():
-    DarkSkyWeather.current(city=Odessa)
-    url = 'https://api.darksky.net/forecast/070cb4f3ace25c5cdfedf89527ba83f6/46.4736019,30.603976'
-    params = {'exclude':'currently,minutely,hourly'}
-    response = requests.get(url, params=params)
+
+class DarkSkyWeather:
+    url = 'https://api.darksky.net/forecast/{api_key}/{coordinates}'
+    params = {}
+    api_key = None
+    city = None
+    city_map = {'odessa': '46.4736019,30.603976'}
+
+    def __init__(self, api_key, city='odessa', params='currently,minutely,hourly'):
+        self.__check_city(city)
+        self.set_api_key(api_key)
+        self.city = city
+        self.__check_params(params)
+
+    def set_api_key(self, api_key):
+        self.api_key = api_key
+
+    def execute_request(self, api_key, coordinates, params):
+        url = self.url.format(api_key=api_key, coordinates=coordinates)
+        response = requests.get(url, params=params)
+        return response
+
+    def __check_city(self, city):
+        assert self.city_map.get(city) is not None, 'No such city in list!'
```

Comment:
`self.city_map.get(city) is not None` could be replaced for `if self.city_map.get(city)`